feat(cli): log attempts to save url map

chosak opened the pull request:
Related to #250.

patrickhulce commented on packages/cli/src/upload/upload.js:
```diff
@@ -342,20 +342,33 @@ async function getPreviousUrlMap(options) {
  * @return {Promise<void>}
  */
 async function writeUrlMapToApi(urlMap) {
+  print(`Saving URL map to temporary public storage...\n`);
```
this feels a bit excessive considering we don't even print that much for the github intergration which is much more important. one-line about its status feels enough :)
```suggestion
```

patrickhulce commented on packages/cli/src/upload/upload.js:
```diff
@@ -342,20 +342,33 @@ async function getPreviousUrlMap(options) {
-  if (!slug) return;
+
+  if (!slug) {
+    print(`No GitHub repository slug found, skipping.\n`);
```
```suggestion
    print(`No GitHub repository slug found, skipping URL map upload.\n`);
```

patrickhulce commented on packages/cli/src/upload/upload.js:
```diff
@@ -342,20 +342,33 @@ async function getPreviousUrlMap(options) {
+    print(`No GitHub repository slug found, skipping.\n`);
+    return;
+  }
+  print(`Saving URL map under GitHub repository slug ${slug}...`);
```
```suggestion
  print(`Saving URL map for GitHub repository ${slug}...`);
```

WDYT? :)

patrickhulce commented on packages/cli/src/upload/upload.js:
```diff
@@ -342,20 +342,33 @@ async function getPreviousUrlMap(options) {
+    }
   } catch (err) {
-    print(`Failed to save urlMap: ${err.message}`);
+    print(`Failed to save URL map: ${err.message}`);
```
```suggestion
    print(`Failed to save URL map: ${err.message}\n`);
```

missed this last time thanks for the fix!

chosak commented:
I see why the tests are failing; because I've configured them to check for output from an upload, but that doesn't happen on when GHA runs on non-`master` checks. @patrickhulce what's optimal behavior here -- to explicitly disable the upload (and roll back test changes), or maybe hardcode an explicit slug to use in the environment [here](https://github.com/GoogleChrome/lighthouse-ci/blob/1bf2e963766214111787d616ef45f198912cfb79/packages/cli/test/autorun-static-dir.test.js#L21)?

patrickhulce commented:
> maybe hardcode an explicit slug to use in the environment here?

That seems like the best path forward, hardcode the use of `GoogleChrome/lighthouse-ci-unit-tests` as the slug and force the `--uploadUrlMap` flag?

chosak commented:
Thanks for the text suggestions, I've fixed and rebased. I'll address your slug suggestion.

Aside: While doing that I noticed that some uses of "GitHub" (for example [`getGitHubRepoSlug`](https://github.com/GoogleChrome/lighthouse-ci/blob/1bf2e963766214111787d616ef45f198912cfb79/packages/utils/src/build-context.js#L315), or the logging statements I've added)  are perhaps overly specific, as one might potentially be running Lighthouse against repositories hosted on other source code providers. But maybe that's outside the scope of this PR.

patrickhulce commented:
Those names are explicit because they're used for things that are github support only. We don't support any other providers and I consider it a bug that they can pickup anything other than github slugs :)

chosak commented:
Thanks @patrickhulce!

Interesting side note I discovered while working on the tests -- some of [these](https://github.com/GoogleChrome/lighthouse-ci/blob/1bf2e963766214111787d616ef45f198912cfb79/packages/utils/test/build-context.test.js#L174) tests won't pass locally if your Git `origin` remote doesn't point to this repository (for example, I often use `upstream` for a project and then `origin` for my fork).

patrickhulce commented:
> some of these tests won't pass locally if your Git origin remote doesn't point to this repository (for example, I often use upstream for a project and then origin for my fork).

Ah yeah thanks for the note I'll loosen those! :)
